calyx-py/test/correctness/queues/fifo.futil

```diff
@@ -0,0 +1,280 @@
+import "primitives/core.futil";
+import "primitives/memories/seq.futil";
+import "primitives/binary_operators.futil";
+component fifo(cmd: 2, value: 32) -> () {
+  cells {
+    mem = seq_mem_d1(32, 16, 4);
+    reg_1 = std_reg(4);
+    reg_2 = std_reg(4);
+    ref ans = std_reg(32);
+    ref err = std_reg(1);
+    reg_3 = std_reg(5);
+    eq_4 = std_eq(5);
+    reg_2_incr_1_5 = std_add(4);
+    reg_3_decr_1_6 = std_sub(5);
+    eq_7 = std_eq(5);
+    eq_8 = std_eq(5);
+    reg_1_incr_1_9 = std_add(4);
+    reg_3_incr_1_10 = std_add(5);
+    cmd_eq_0 = std_eq(2);
+    cmd_eq_1 = std_eq(2);
+    cmd_eq_2 = std_eq(2);
+    cmd_eq_3 = std_eq(2);
+  }
+  wires {
+    group raise_err {
+      err.in = 1'd1;
+      err.write_en = 1'd1;
+      raise_err[done] = err.done;
+    }
+    comb group eq_4_group {
+      eq_4.left = reg_3.out;
+      eq_4.right = 5'd0;
+    }
+    group read_payload_from_mem_pop {
+      mem.addr0 = reg_2.out;
+      mem.content_en = 1'd1;
+      ans.write_en = mem.done ? 1'd1;
+      ans.in = mem.done ? mem.read_data;
+      read_payload_from_mem_pop[done] = ans.done;
+    }
+    group reg_2_incr_1_5_group {
+      reg_2_incr_1_5.left = reg_2.out;
+      reg_2_incr_1_5.right = 4'd1;
+      reg_2.write_en = 1'd1;
+      reg_2.in = reg_2_incr_1_5.out;
+      reg_2_incr_1_5_group[done] = reg_2.done;
+    }
+    group reg_3_decr_1_6_group {
+      reg_3_decr_1_6.left = reg_3.out;
+      reg_3_decr_1_6.right = 5'd1;
+      reg_3.write_en = 1'd1;
+      reg_3.in = reg_3_decr_1_6.out;
+      reg_3_decr_1_6_group[done] = reg_3.done;
+    }
+    comb group eq_7_group {
+      eq_7.left = reg_3.out;
+      eq_7.right = 5'd0;
+    }
+    group read_payload_from_mem_peek {
+      mem.addr0 = reg_2.out;
+      mem.content_en = 1'd1;
+      ans.write_en = mem.done ? 1'd1;
+      ans.in = mem.done ? mem.read_data;
+      read_payload_from_mem_peek[done] = ans.done;
+    }
+    comb group eq_8_group {
+      eq_8.left = reg_3.out;
+      eq_8.right = 5'd16;
+    }
+    group write_payload_to_mem {
+      mem.addr0 = reg_1.out;
+      mem.write_en = 1'd1;
+      mem.write_data = value;
+      write_payload_to_mem[done] = mem.done;
+      mem.content_en = 1'd1;
+    }
+    group reg_1_incr_1_9_group {
+      reg_1_incr_1_9.left = reg_1.out;
+      reg_1_incr_1_9.right = 4'd1;
+      reg_1.write_en = 1'd1;
+      reg_1.in = reg_1_incr_1_9.out;
+      reg_1_incr_1_9_group[done] = reg_1.done;
+    }
+    group reg_3_incr_1_10_group {
+      reg_3_incr_1_10.left = reg_3.out;
+      reg_3_incr_1_10.right = 5'd1;
+      reg_3.write_en = 1'd1;
+      reg_3.in = reg_3_incr_1_10.out;
+      reg_3_incr_1_10_group[done] = reg_3.done;
+    }
+    cmd_eq_0.left = cmd;
+    cmd_eq_0.right = 2'd0;
+    cmd_eq_1.left = cmd;
+    cmd_eq_1.right = 2'd1;
+    cmd_eq_2.left = cmd;
+    cmd_eq_2.right = 2'd2;
+    cmd_eq_3.left = cmd;
+    cmd_eq_3.right = 2'd3;
+  }
+  control {
+    par {
+      par {
```

Comment:
This control was generated using a `case` helper, and it double-wrapped in `par`.